### PR TITLE
Fix layered overlay rendering and highlight

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/8296/Layered-drawInFront-drawHUDLayer-Model-Overlays-don-t-render-properly-and-then-crash-on-layer-switch
https://highfidelity.manuscript.com/f/cases/12742/Overlay-in-front-drawn-over-highlighted-object-disappears-because-of-the-highlight-stencil
https://highfidelity.manuscript.com/f/cases/14923/In-VR-lasers-are-hidden-by-highlighted-grabbable-objects-3D-bounding-boxes

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/c4c8420dee8acf8608ca1db409fa9f6d/raw/21d71bf4e0cf42c38b88216030d29f787e3f33a6/ModelOverlayEntityTest.js).  You should see two rows of models.  The bottom row is entities, and the top row is overlays.  They should look identical, like this:
hifi-snap-by-samgondelman-on-2018-06-06_10-46-42

- Press "u".  The top layer should render "in front" now (for example, in third person, they'll render on top of your avatar).  Besides that, they should still look identical to the bottom row, like this:
hifi-snap-by-samgondelman-on-2018-06-06_10-46-55

- Press "i".  The top layer will render in the HUD layer now (for example, if you bring up the stats ("/"), they'll render on top of the 2D overlay).  They'll look darker and not-aliased, but that's ok.  Confirm that they look correct otherwise, like this:
hifi-snap-by-samgondelman-on-2018-06-06_10-47-04

- Run [this](https://gist.githubusercontent.com/SamGondelman/5f1dcb5f65d999bc1fdabdb6bb58b147/raw/2532d2844a1f3bdfa9cbcbec881d59df539086d4/LayerOverlayTest.js).  You should see this:
hifi-snap-by-samgondelman-on-2018-06-06_10-48-05

- Create a grabbable box.  In VR, when you highlight it with your hand lasers, the lasers should continue rendering properly in front of it.  NOT like this:
https://media.giphy.com/media/3gWxE7QRTlFNaLbYbn/giphy.gif

- Confirm that things still render properly on Android.  Test a few domains.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10378/HighFidelity-Beta-PR13312-d2f8ee3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10378/HighFidelity-Beta-PR13312-d2f8ee3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10378/HighFidelity-Beta-Interface-PR13312-d2f8ee3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10378/HighFidelity-Beta-PR13312-d2f8ee3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10378/HighFidelity-Beta-Interface-PR13312-d2f8ee3.exe

**Comment by samcake** [comment]
Delaying  to 70
No time to dev approved

**Comment by Qliemillar** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10655/HighFidelity-Beta-PR13312-d2f8ee3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10655/HighFidelity-Beta-PR13312-d2f8ee3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10655/HighFidelity-Beta-Interface-PR13312-d2f8ee3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10655/HighFidelity-Beta-PR13312-d2f8ee3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10655/HighFidelity-Beta-Interface-PR13312-d2f8ee3.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Overlays on Windows 10 look good, Android Smoke passes.

File: libraries/render-utils/src/RenderPipelines.cpp [hunk 1]
```diff
@@ -501,7 +442,7 @@ void addPlumberPipeline(ShapePlumber& plumber,
         bool isWireframed = (i & 4);
 
         auto state = std::make_shared<gpu::State>();
-        PrepareStencil::testMaskDrawShape(*state);
+        key.isTranslucent() ? PrepareStencil::testMask(*state) : PrepareStencil::testMaskDrawShape(*state);
```
[the remark this answers is not in the document]
Comment (SamGondelman) [reply libraries/render-utils/src/RenderPipelines.cpp]: This is what we have for getSimplePipeline:
```
    if (config.isAntiAliased()) {
        config.isTransparent() ? PrepareStencil::testMask(*state) : PrepareStencil::testMaskDrawShape(*state);
    } else {
        PrepareStencil::testMaskDrawShapeNoAA(*state);
    }
```

**Pull request merged**